website/blog/2018-07-14-whats-the-deal-with-the-new-pipeline-plugin-configuration.md

```diff
@@ -0,0 +1,153 @@
+  |> # + '!'
+  |> new User.Message(#)
+  |> await stream.write(#)
+  |> console.log;
+```
+
+In ["topic style"](https://github.com/js-choi/proposal-smart-pipelines/blob/master/readme.md#topic-style), a placeholder (called a "lexical topic token") is required, and the code will throw an early SyntaxError if it is not included.
+
+```js
+10 |> # + 1;
+promise |> await #;
+```
+
+In ["bare style"](https://github.com/js-choi/proposal-smart-pipelines/blob/master/readme.md#bare-style), where a bare identifier is provided to a step in the pipeline, no token is necessary.
+
+```js
+x |> a;
+x |> f.b;
+```
+
+If there are any operators, parentheses (including for method calls), brackets, or anything other than identifiers and dot punctuators, then it is in topic style, not in bare style, and a topic token is necessary. This is avoids footguns and eliminate ambiguity in bare style code.
+
+Smart pipelines thus resolve the issue of async in an integrative way, allowing all possible expressions to be emedded in a pipeline; not only `await`, but also `typeof`, `yield`, and another other operator desired.
+
+## Where Babel Comes In
+
+Once all three proposals were fleshed out, we realized discussion and debate were unlikely to resolve the inherent tension between them. We decided the best way to make a decision would be with the feedback of developers, using the proposals in real code. Given babel's role in the community, we approached them about including all three of the above proposals into the pipeline operator plugin.
+
+Because these proposals parse slightly differently, support needs to be added to `@babel/parser` (formerly `babylon`), and the parser needs to be configured based on which proposal is its current parse target. The pipeline operator plugin itself thus needs the `"proposal"` option, both to configure babylon as well as its own transformation.
+
+We were working under a compressed timeline for this, as we needed to introduce any breaking changes to babel, `@babel/parser`, as well as the pipeline proposal plugin before babel@7 left beta. We also wanted the plugin to be able to eventually default to whichever of the proposals get accepted, so the configuration option becomes obsolete. Given these two constraints, we opted to introduce this new configuration option and make it required, forcing users to decide which proposal they want to use in their project.
+
+Because the pipeline proposal is currently stage-1, both the `stage-1` & `stage-0` plugins include the pipeline operator. An option was added to both of those plugins, `"pipelineProposal"`, to forward the value to the pipeline plugin. This option was also required, and thus breaking, so it caused some frustration for some users. The stage presets have been deprecated and [are being removed](https://github.com/babel/babel/pull/8293), because when those presets are used, consumers often don't know what plugins they're including, so changes like this are surprising and frustrating.
```

Comment:
Yeah in this section we can explain this is the approach we'd like to take moving forward for any kind of proposal that doesn't have a clear semantics moving forward (like Stage 0-2). Where we might force an opt-in to an option vs. some default since that implies it's ready.

Reply:
So happy about this approach and creating good documentation for it!